File: Two Sum/Two Sum.js
let nums = [3,3];
let target = 6;

// My solution 
var twoSum = function(nums, target) {
    for (let index_i in nums) {
        for(let index_j in nums) {
            if (index_i === index_j) continue;
            if (target == nums[index_i] + nums[index_j]) {
                return [index_i, index_j]
            }
        }
    }
};

// Better solution
function twoSum1(nums, target) {
    let hashmap = {};
    for (let i in nums) {
        hashmap[nums[i]] = i;
    }
    for (let i in nums) {
        let complement = target - nums[i];
        if (complement in hashmap && hashmap[complement] != i) {
            return [i, hashmap[complement]];
        }
    }
}

console.log(twoSum1(nums, target));

File: Two Sum/Two Sum.py
nums = [2, 7, 11, 15]
target = 9

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numsMap = {}
        for i in range(len(nums)):
            numsMap[nums[i]] = i
            if target - nums[i] in numsMap:
                return [numsMap[target - nums[i]], i]